src/accessible_name.ts

```diff
@@ -1,11 +1,19 @@
-import {computeTextAlternative} from './lib/compute_text_alternative';
+import {computeTextAlternative, Rule} from './lib/compute_text_alternative';
 
 /**
- * Main exported function for the library. Initialises traversal with an empty
- * context.
- * @param elem - The element whose accessible name will be calculated
- * @return - The accessible name for elem
+ * Compute the accessible name for a given HTMLElement.
+ * @param elem - The HTMLElement whose accessible name will be calculated.
  */
 export function getAccessibleName(elem: HTMLElement): string {
+  return computeTextAlternative(elem).name;
+}
+
+/**
+ * Get details surrounding the computation of the accessible name for a given HTMLElement
+ * @param elem - The HTMLElement whose accessible name will be calculated.
+ */
+export function getNameComputationDetails(
+  elem: HTMLElement
+): {name: string; nodesUsed: Set<Node>; rulesApplied: Set<Rule>} {
```

Comment:
I was actually considering leaving the calculation of `visitedHTMLSnippet` to the validation app, as the code would be pretty much exactly the same as is written for Chrome at the moment so I could use a single function for both as opposed to copying it across. Also it might be a bit niche to return the `visitedHTMLSnippet` in this function, what do you think?

Also I can add a `ComputationDetails` return type if you think that would be better than just returning the object?